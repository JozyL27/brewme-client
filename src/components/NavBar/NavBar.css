ul {
    display: flex;
    justify-content: center;
    align-items: center;
    margin-top: 20px;
}

.navLink {
    list-style: none;
    margin-right: 15px;
    text-decoration: none;
    text-align: center;
}

.navBarLink {
    color: #E1C340;
    font-weight: bold;
    transition-duration: 0.4s;
}

.navBarLink:hover {
    color: rgb(135,206,250);
    text-shadow: 1px 1px 2px black, 0 0 25px blue, 0 0 5px rgb(135,206,250);
}



@media screen and (min-width: 768px) {
    .navLink {
        font-size: 25px;
    }
}

/* laptops */
@media screen and (min-width: 1024px) {
    .navLink {
        margin-right: 45px;
    }
}