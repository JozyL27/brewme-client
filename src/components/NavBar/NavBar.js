import React from 'react'
import { Link } from 'react-router-dom'
import './NavBar.css'


export default class NavBar extends React.Component {
    render() {
        return (
            <ul className='navBar'>
                <li className='navLink'>
                    <Link
                    to='/'
                    className="navBarLink active">
                    Home
                    </Link>
                </li>
                <li className='navLink'>
                    <Link
                    to='/search'
                    className="navBarLink"
                    >
                    Search
                    </Link>
                </li>
                <li className='navLink'>
                    <Link
                    to='/myBeers'
                    className="navBarLink"
                    >
                    My Beers
                    </Link>
                </li>
                <li className='navLink'>
                    <Link
                    to='/random'
                    className="navBarLink"
                    >
                    Random Beer
                    </Link>
                </li>
            </ul>
        )
    }
}