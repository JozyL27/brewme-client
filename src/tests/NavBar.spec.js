import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import NavBar from '../components/NavBar/NavBar'
import { BrowserRouter } from 'react-router-dom'


describe(`NavBar component`, () => {

    it(`renders without crashing`, () => {
        const div = document.createElement('div')
        ReactDOM.render(
        <BrowserRouter>
        <NavBar />
        </BrowserRouter>
        , div)
        ReactDOM.unmountComponentAtNode(div)
    })

    it(`should render the UI as expected`, () => {
        const tree = renderer
            .create(
            <BrowserRouter>
            <NavBar />
            </BrowserRouter>
            )
            .toJSON()
            expect(tree).toMatchSnapshot()
    })
})