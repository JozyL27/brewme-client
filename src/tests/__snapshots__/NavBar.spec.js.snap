// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar component should render the UI as expected 1`] = `
<ul
  className="navBar"
>
  <li
    className="navLink"
  >
    <a
      className="navBarLink active"
      href="/"
      onClick={[Function]}
    >
      Home
    </a>
  </li>
  <li
    className="navLink"
  >
    <a
      className="navBarLink"
      href="/search"
      onClick={[Function]}
    >
      Search
    </a>
  </li>
  <li
    className="navLink"
  >
    <a
      className="navBarLink"
      href="/myBeers"
      onClick={[Function]}
    >
      My Beers
    </a>
  </li>
  <li
    className="navLink"
  >
    <a
      className="navBarLink"
      href="/random"
      onClick={[Function]}
    >
      Random Beer
    </a>
  </li>
</ul>
`;
